code/game/objects/items/toys.dm

```diff
@@ -1086,3 +1087,182 @@
+			to_chat(user, "You can't blow up \the [src] without lungs!")
+			return
+	inflate(user)
+
+/obj/item/toy/balloon/proc/inflate(mob/user, datum/gas_mixture/G)
+	var/obj/item/toy/balloon/inflated/B = new(get_turf(src), col)
+	if(user)
+		user.drop_item(src, force_drop = 1)
+		user.put_in_hands(B)
+		to_chat(user, "You blow up \the [src].")
+	playsound(src, 'sound/misc/balloon_inflate.ogg', 50, 1)
+	if(!G)
+		B.air_contents = new /datum/gas_mixture()
+		B.air_contents.volume = 12 //liters
+		B.air_contents.temperature = T20C
+		B.air_contents.carbon_dioxide = 0.5
+	else
+		var/moles = ONE_ATMOSPHERE*12/(R_IDEAL_GAS_EQUATION*G.temperature)
+		B.air_contents = G.remove(moles)
+	qdel(src)
+
+/obj/item/toy/balloon/inflated
+	desc = "An inflated balloon. You have an urge to pop it."
+	icon_state = "balloon"
+	w_class = W_CLASS_MEDIUM
+	var/datum/gas_mixture/air_contents = null
+	var/can_be_strung = TRUE
+
+/obj/item/toy/balloon/inflated/attack_self(mob/user)
+	return
+
+/obj/item/toy/balloon/inflated/attackby(obj/item/weapon/W, mob/user)
+	if(W.sharpness_flags & (SHARP_TIP|HOT_EDGE))
+		user.visible_message("<span class='warning'>\The [user] pops \the [src]!</span>","You pop \the [src].")
+		pop()
+		return
+	if(istype(W, /obj/item/stack/cable_coil) && can_be_strung)
+		var/obj/item/stack/cable_coil/C = W
+		C.use(1)
+		to_chat(user, "You tie some of \the [C] around the end of \the [src].")
+		new /obj/item/toy/balloon/inflated/string(get_turf(src), col)
+		qdel(src)
+
+/obj/item/toy/balloon/inflated/proc/pop()
+	playsound(src, 'sound/misc/balloon_pop.ogg', 100, 1)
+	if(air_contents)
+		loc.assume_air(air_contents)
+	qdel(src)
+
+/obj/item/toy/balloon/inflated/bullet_act()
+	pop()
+
+/obj/item/toy/balloon/inflated/fire_act(datum/gas_mixture/air, exposed_temperature, exposed_volume)
+	if(exposed_temperature > T0C+100)
+		pop()
+
+/obj/item/toy/balloon/inflated/string
+	desc = "An inflated balloon with a string hanging from it. You have an urge to pop it."
+	icon_state = "balloon_with_string"
+	can_be_strung = FALSE
+
+/obj/item/toy/balloon/inflated/string/update_icon()
+	..()
+	var/image/balleft = image('icons/mob/in-hand/left/toys.dmi', src, "[icon_state]")
+	var/image/balleftshadow = image('icons/mob/in-hand/left/toys.dmi', src, "[icon_state]_shadow")
+	var/image/balright = image('icons/mob/in-hand/right/toys.dmi', src, "[icon_state]")
+	var/image/balrightshadow = image('icons/mob/in-hand/right/toys.dmi', src, "[icon_state]_shadow")
+	balleft.icon += col
+	balright.icon += col
+	balleft.overlays += balleftshadow
+	balright.overlays += balrightshadow
+	dynamic_overlay["[HAND_LAYER]-[GRASP_LEFT_HAND]"] = balleft
+	dynamic_overlay["[HAND_LAYER]-[GRASP_RIGHT_HAND]"] = balright
+
+/obj/item/toy/balloon/glove
+	name = "latex glove"
+	desc = "A latex glove."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "latexballoon"
+	item_state = "lgloves"
+	col = null
+
+/obj/item/toy/balloon/glove/inflate(mob/user, datum/gas_mixture/G)
```

Comment:
If this is just copy-pasted for the glove spawning, you can make that a type var or a proc.